src/overlays/actors/ovl_En_Encount3/z_en_encount3.c

```diff
@@ -32,22 +33,174 @@ ActorInit En_Encount3_InitVars = {
+    if ((this->unk15C >= 0) && (Flags_GetSwitch(play, this->unk15C) != 0)) {
+        Actor_Kill(&this->actor);
+    }
+    this->actor.flags |= 0x08000000;
+    this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+    func_809AD058(this);
+}
+
+void EnEncount3_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_809AD058(EnEncount3* this) {
+    this->unk154 = 0x113;
+    this->unk150 = 1;
+    this->unk15E = 0x1E;
+    this->actionFunc = func_809AD084;
+}
+
+void func_809AD084(EnEncount3* this, PlayState* play) {
+    if ((this->unk15C >= 0) && (Flags_GetSwitch(play, this->unk15C) != 0)) {
+        Actor_Kill(&this->actor);
+    } else if (!(this->unk16C < this->actor.xzDistToPlayer) && (Player_GetMask(play) == 6) && (!D_809AD810)) {
+        if (this->unk15E > 0) {
+            this->unk15E--;
+        } else {
+            this->unk17C =
+                Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, this->unk154, this->actor.world.pos.x,
+                                   this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, this->unk158);
+            if (this->unk17C != NULL) {
+                this->unk14E++;
+                D_809AD810 = true;
+                this->actionFunc = func_809AD194;
+            }
+        }
+    }
+}
+
+void func_809AD194(EnEncount3* this, PlayState* play) {
+    if (this->unk14E == 0) {
+        this->unk178 = 0.0f;
+        if (this->unk15C >= 0) {
+            Flags_SetSwitch(play, this->unk15C);
+        }
+        this->actionFunc = func_809AD1EC;
+    }
+}
+
+void func_809AD1EC(EnEncount3* this, PlayState* play) {
+    if (this->unk174 < 0.002f) {
+        D_809AD810 = false;
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnEncount3_Update(Actor* thisx, PlayState* play2) {
+    EnEncount3* this = THIS;
+    f32 new_var;
+    PlayState* play = play2;
+    Player* player = GET_PLAYER(play);
+    f32 sp3C;
+    f32 sp38;
+
+    this->actionFunc(this, play);
+    if (this->actionFunc == func_809AD194) {
+        new_var = this->unk16C * 0.52f;
+        if (new_var < this->actor.xzDistToPlayer) {
+            this->unk178 = ((this->unk16C * 0.03125f) * 0.001f) + 0.1f;
+        } else {
+            this->unk178 = 0.06f;
+        }
+
+        if (((this->unk16C + 50.0f) + BREG(0)) < this->actor.xzDistToPlayer) {
+            if (this->actionFunc == func_809AD194) {
+                this->unk148 = 0;
+                this->unk178 = 0.0f;
+                D_809AD810 = false;
+                if (((this->unk17C != NULL) && (this->unk17C->update != NULL)) &&
+                    (this->unk17C->colChkInfo.health > 0)) {
+                    Actor_Kill(this->unk17C);
+                    this->unk17C = NULL;
+                }
+                func_809AD058(this);
+            }
+        } else if (this->unk16C < this->actor.xzDistToPlayer) {
+            s16 i = 0;
+            while (i < PLAYER_BODYPART_MAX) {
+                player->flameTimers[i++] = Rand_S16Offset(0, 0xC8);
+            }
```

Comment:
Can you change it to a for instead?
```suggestion
            s16 i;

            for (i = 0; i < PLAYER_BODYPART_MAX; i++) {
                player->flameTimers[i] = Rand_S16Offset(0, 0xC8);
            }
```